Flow.Launcher/Helper/HotKeyMapper.cs

```diff
@@ -0,0 +1,110 @@
+using NHotkey.Wpf;
+using Flow.Launcher.Core.Resource;
+using System.Windows;
+using Flow.Launcher.ViewModel;
+
+namespace Flow.Launcher.Helper
+{
+    internal static class HotKeyMapper
+    {
+        private static Settings settings;
+        private static MainViewModel mainViewModel;
+
+        internal static void Initialize(MainViewModel mainVM)
+        {
+            mainViewModel = mainVM;
+            settings = mainViewModel._settings;
+
+            SetHotkey(settings.Hotkey, OnHotkey);
+            SetCustomPluginHotkey();
+        }
+
+        private static void SetHotkey(string hotkeyStr, EventHandler<HotkeyEventArgs> action)
+        {
+            var hotkey = new HotkeyModel(hotkeyStr);
+            SetHotkey(hotkey, action);
+        }
+
+        internal static void SetHotkey(HotkeyModel hotkey, EventHandler<HotkeyEventArgs> action)
+        {
+            string hotkeyStr = hotkey.ToString();
+            try
+            {
+                HotkeyManager.Current.AddOrReplace(hotkeyStr, hotkey.CharKey, hotkey.ModifierKeys, action);
+            }
+            catch (Exception)
+            {
+                string errorMsg =
+                    string.Format(InternationalizationManager.Instance.GetTranslation("registerHotkeyFailed"),
+                        hotkeyStr);
+                MessageBox.Show(errorMsg);
+            }
+        }
+
+        internal static void RemoveHotkey(string hotkeyStr)
+        {
+            if (!string.IsNullOrEmpty(hotkeyStr))
+            {
+                HotkeyManager.Current.Remove(hotkeyStr);
+            }
+        }
+
+        internal static void OnHotkey(object sender, HotkeyEventArgs e)
+        {
+            if (!ShouldIgnoreHotkeys())
+            {
+                if (settings.LastQueryMode == LastQueryMode.Empty)
+                {
+                    mainViewModel.ChangeQueryText(string.Empty);
+                }
+                else if (settings.LastQueryMode == LastQueryMode.Preserved)
+                {
+                    mainViewModel.LastQuerySelected = true;
+                }
+                else if (settings.LastQueryMode == LastQueryMode.Selected)
+                {
+                    mainViewModel.LastQuerySelected = false;
+                }
+                else
+                {
+                    throw new ArgumentException($"wrong LastQueryMode: <{settings.LastQueryMode}>");
+                }
+
+                mainViewModel.ToggleFlowLauncher();
+                e.Handled = true;
+            }
+        }
+
+        /// <summary>
+        /// Checks if Flow Launcher should ignore any hotkeys
+        /// </summary>
+        /// <returns></returns>
+        private static bool ShouldIgnoreHotkeys()
+        {
+            //double if to omit calling win32 function
+            if (settings.IgnoreHotkeysOnFullscreen)
+                if (WindowsInteropHelper.IsWindowFullscreen())
+                    return true;
+
+            return false;
```

Comment:
Also could you change this to short circuit?

```c#
return settings.IgnoreHotkeysOnFullscreen && WindowsInteropHelper.IsWindowFullscreen();
```

It will omit calling win32 function when the first one is false.

Reply:
it's same thing isnt it, current code wont call the win32 function anyway if settings.IgnoreHotkeysOnFullscreen is false

Reply:
Yeah I know. Just it looks cleaner, and the comment is a bit confusing suggesting that we need the double if to do so.